static/app/components/profiling/FlamegraphZoomViewMinimap.tsx

```diff
@@ -0,0 +1,462 @@
+        vec2.create(),
+        vec2.fromValues(-1, -1),
+        configCenter
+      );
+
+      const translated = mat3.translate(mat3.create(), identity, configCenter);
+      const scaled = mat3.scale(mat3.create(), translated, vec2.fromValues(scale, 1));
+      const translatedBack = mat3.translate(mat3.create(), scaled, invertedConfigCenter);
+
+      canvasPoolManager.dispatch('transformConfigView', [translatedBack]);
+    },
+    [flamegraphMiniMapRenderer, canvasPoolManager]
+  );
+
+  const onMinimapCanvasMouseDown = React.useCallback(
+    evt => {
+      if (!configSpaceCursor || !flamegraphMiniMapRenderer || !canvasPoolManager) {
+        return;
+      }
+
+      const logicalMousePos = vec2.fromValues(
+        evt.nativeEvent.offsetX,
+        evt.nativeEvent.offsetY
+      );
+      const physicalMousePos = vec2.scale(
+        vec2.create(),
+        logicalMousePos,
+        window.devicePixelRatio
+      );
+
+      if (
+        flamegraphConfigViewRef.current.contains(vec2.fromValues(...configSpaceCursor))
+      ) {
+        setLastDragVector(physicalMousePos);
+      } else {
+        const startConfigSpaceCursor = flamegraphMiniMapRenderer.getConfigSpaceCursor(
+          vec2.fromValues(evt.nativeEvent.offsetX, evt.nativeEvent.offsetY),
+          flamegraphMiniMapRenderer.configSpaceToPhysicalSpace
+        );
+        setStartDragConfigSpaceCursor(startConfigSpaceCursor);
+      }
+    },
+    [configSpaceCursor, flamegraphMiniMapRenderer, canvasPoolManager]
+  );
+
+  const onMinimapCanvasMouseUp = React.useCallback(() => {
+    setConfigSpaceCursor(null);
+    setStartDragConfigSpaceCursor(null);
+    setLastDragVector(null);
+  }, []);
+
+  React.useEffect(() => {
+    if (!flamegraphMiniMapCanvasRef) {
+      return undefined;
+    }
+
+    const onCanvasWheel = (evt: WheelEvent) => {
+      evt.preventDefault();
+      const isZoom = evt.metaKey;
+
+      // @TODO figure out what key to use for other platforms
+      if (isZoom) {
+        onMinimapZoom(evt);
+      } else {
+        onMinimapScroll(evt);
+      }
+    };
+
+    flamegraphMiniMapCanvasRef.addEventListener('wheel', onCanvasWheel);
+    return () => flamegraphMiniMapCanvasRef?.removeEventListener('wheel', onCanvasWheel);
+  }, [flamegraphMiniMapCanvasRef, onMinimapScroll, onMinimapZoom]);
+
+  return (
+    <React.Fragment>
+      <canvas
+        ref={canvas => setFlamegraphMiniMapRef(canvas)}
+        onMouseDown={onMinimapCanvasMouseDown}
+        onMouseMove={onMinimapCanvasMouseMove}
+        onMouseLeave={onMinimapCanvasMouseUp}
+        style={{
+          width: '100%',
+          height: '100%',
+          position: 'absolute',
+          left: 0,
+          top: 0,
+          userSelect: 'none',
+          cursor:
+            configSpaceCursor &&
+            flamegraphConfigViewRef.current.contains(
+              vec2.fromValues(...configSpaceCursor)
+            )
+              ? 'grab'
+              : 'col-resize',
+        }}
```

Comment:
Reminder we typically avoid inlining styles

Reply:
Oh yeah, I keep forgetting...